### PR TITLE
adds the ACL token to consul client when fetching catalog services

### LINKED ISSUE #201: Support ACL for Service Discovery
Since 0.6.0

See http://stackoverflow.com/questions/38064011/spring-cloud-consul-discovery-acl-for-catalog-services

### PR DESCRIPTION
**Pull request by xetys**
fix #201

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@xetys Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-consul&pullRequestId=233)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-consul&pullRequestId=233) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@xetys Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-consul&pullRequestId=233)!

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClient.java [hunk 1]
```diff
@@ -163,7 +163,14 @@ private void addInstancesToList(List<ServiceInstance> instances, String serviceI
 
 	@Override
 	public List<String> getServices() {
-		return new ArrayList<>(client.getCatalogServices(QueryParams.DEFAULT).getValue()
-				.keySet());
+		String aclToken = properties.getAclToken();
+
+		if (aclToken != null && !aclToken.isEmpty()) {
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClient.java]: Use `StringUtils.hasText(aclToken)` from spring `util` package.

**Comment by xetys** [comment]
the last commit contains now a addtional test, but also performs little changes on the ACL setup. Giving "anonymous" write rights on all services (as it was before) is higher rated than the limitation of the discovery client acl. With the current setup now the additional test would fail, if the discovery client would not be aware of configured consul tokens.

**Comment by xetys** [comment]
may I ask whether this state is ok for now or still in review?

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClient.java [hunk 2]
```diff
@@ -141,8 +141,16 @@ private String getAgentHost() {
 
 	private void addInstancesToList(List<ServiceInstance> instances, String serviceId,
 			QueryParams queryParams) {
-		Response<List<HealthService>> services = client.getHealthServices(serviceId,
-				this.properties.isQueryPassing(), queryParams);
+
+		String aclToken = properties.getAclToken();
+        Response<List<HealthService>> services;
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClient.java]: replace tabs with spaces

**Comment by xetys** [comment]
sorry if it's annoying, but is here still something to do for me? or can this be merged?

**Comment by spencergibb** [comment]
@xetys getting back from holiday. lt's in my queue to review.

**Comment by spencergibb** [comment]
looks like there was a test failure.

**Comment by xetys** [comment]
yeah, I noticed that, but had no time to fix that....I will do this the next days...hope this then gets merged :)

**Comment by spencergibb** [comment]
@xetys yup, I'm hoping to clear the queue this week, so if you can get to it, I'll get it merged soon.

**Comment by xetys** [comment]
@spencergibb I am sorry for the long delay...it took me some time to get back to my open tasks

**Comment by xetys** [comment]
```
[INFO] spring-cloud-consul-dependencies ................... SUCCESS [  0.003 s]
[INFO] Spring Cloud Consul ................................ SUCCESS [  1.735 s]
[INFO] Spring Cloud Consul Core ........................... SUCCESS [  2.608 s]
[INFO] Spring Cloud Consul Config ......................... SUCCESS [ 17.796 s]
[INFO] Spring Cloud Consul Discovery ...................... SUCCESS [01:16 min]
[INFO] Spring Cloud Consul Binder ......................... SUCCESS [01:02 min]
[INFO] Spring Cloud Starter Consul ........................ SUCCESS [  0.589 s]
[INFO] Spring Cloud Starter Consul Bus .................... SUCCESS [  0.925 s]
[INFO] Spring Cloud Starter Consul Config ................. SUCCESS [  0.513 s]
[INFO] Spring Cloud Starter Consul Discovery .............. SUCCESS [  1.074 s]
[INFO] Spring Cloud Starter Consul All .................... SUCCESS [  1.203 s]
[INFO] Spring Cloud Consul Sample ......................... SUCCESS [  1.256 s]
[INFO] Spring Cloud Consul Docs ........................... SUCCESS [  0.230 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:48 min
[INFO] Finished at: 2017-02-20T02:23:26+01:00
[INFO] Final Memory: 134M/689M
[INFO] ------------------------------------------------------------------------
```

am I missing something?

**Comment by spencergibb** [comment]
Squashed and rebased with test fixes via e6cbd11a03b22469aadaa90c1ade5d6cb3ca3770

**Pull request closed without merging**